api/routers/areas.py

```diff
@@ -201,14 +259,19 @@ def get_area_occupancy_rules(area_id: str):
     area_config_path = area.get_config_path()
 
     if not os.path.exists(area_config_path):
-        return OccupancyRuleListDTO.parse_obj([])
+        return OccupancyRuleListDTO.parse_obj([]).__root__
 
     with open(area_config_path, "r") as area_file:
         rules_data = json.load(area_file)
     return OccupancyRuleListDTO.from_store_json(rules_data)
 
 
 def set_occupancy_rules(area_id: str, rules):
+    """
+    Doubt:
+    I don't think these functions are called correctly in the above endpoints.
```

Comment:
I think rules should be a `OccupancyRuleListDTO` if set. What might be wrong is the default value of that field in AreaConfigDTO which I think should either be None or an empty OccupancyRuleListDTO

Reply:
What I was trying to explain is that:
set_occupancy_rules is called with two parameters.
An identification (chain) and rules. If we pay attention to where we are using the rules parameter within the function, we can notice that we call its .to_store_json () method.

```
def set_occupancy_rules(area_id: str, rules):
    """
    Doubt:
    I don't think these functions are called correctly in the above endpoints.
    Rules are lists and lists do not have .to_store_json()
    """
    area_config_path = get_config().get_area_config_path(area_id)
    Path(os.path.dirname(area_config_path)).mkdir(parents=True, exist_ok=True)

    if os.path.exists(area_config_path):
        with open(area_config_path, "r") as area_file:
            data = json.load(area_file)
    else:
        data = {}

    with open(area_config_path, "w") as area_file:
        data["occupancy_rules"] = rules.to_store_json()["occupancy_rules"]
        json.dump(data, area_file)
```

However, when we call the function, "rules" parameter is a list of AreaOccupancyRule, but it is a list. And it does not have the function .to_store_json()
```
    occupancy_rules = edited_area.occupancy_rules
    del edited_area.occupancy_rules

    area_dict = map_to_config_file_format(edited_area)
    config_dict[f"Area_{index}"] = area_dict
    success = update_config(config_dict, reboot_processor)

    if occupancy_rules:
        set_occupancy_rules(edited_area.id, occupancy_rules)
    else:
        delete_area_occupancy_rules(area_id)
```